core/storage/base_model/gae_models.py

```diff
@@ -386,8 +386,6 @@ class BaseCommitLogEntryModel(BaseModel):
 
     # The id of the user.
     user_id = ndb.StringProperty(indexed=True, required=True)
-    # The username of the user, at the time of the edit.
-    username = ndb.StringProperty(indexed=True, required=True)
```

Comment:
I don't think you can remove this yet. For now, just mark it as deprecated. You'll need to run the job, we'll need to verify that this is None for all models, and only then we can remove it.

Otherwise I think we end up with errors in prod (happened before and led to an emergency reinstatement of the just-deleted field).

That said, if you've tested and are sure this won't cause a problem, I'm happy to run the current PR on a prod server and check.

Reply:
The problem is that setting all instances fields to `None` is not enough. In order to properly remove the field, we need to remove it here and then run the one-off job.

So if we decide to first set all `username` fields to `None`, we will need to run this job after anyway.

Reply:
Can you explain the steps you took to manually test that all this will work in practice? Like I said, if we remove a field from a model class and then do a deployment, then if there are existing models in the datastore with that field set, we run into production errors as soon as we try to access existing models. This happens even before we run the job, and it is important that the server still works fine in the interim (assume arbitrary delays between the deployment and when the job is run).

Also, if you've tested manually and are sure that the above concern is not an issue, then please fill in the release jobs form and I'll do a test run in prod.

Reply:
Yep, I did test it on my own instance and there were no interim errors since all the references to `username` are removed and just accessing a model with `username` filled doesn't produce errors because AppEngine tolerates when the models (saved in datastore) have fields that are currently not defined in the code. I have filled the release jobs.

Reply:
I did a test run. It looks good. Thanks!